### PR TITLE
Add StarGeneratorTest.java

### PR DESCRIPTION
**Pull request by jsachetti**
Added StarGeneratorTest.java to src/metrocs/redistricting, tests StarGenerator.java with JUnit
One method, generatePatternTest tests whether a null value is returned from the StarGenerator generatePattern method with inputs of 1 and 4, and tests if 2 patterns with same input are equal.
Ant build successful and no checkstyle errors on our machines.
- #129

### CODE REVIEW

File: src/metrocs/redistricting/StarGeneratorTest.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+ * Tests for Voter objects.
+ *
+ * @author Jada, David, Colin, John, Daniel, Ali
+ * @version
+ */
+public class StarGeneratorTest {
+    @Test
+    public void generatePatternTest() {
+        ArrayList<ArrayList<Location>> firstLocation = new ArrayList<ArrayList<Location>>();
+        ArrayList<ArrayList<Location>> secondLocation = new ArrayList<ArrayList<Location>>();
```
Comment (jody) [inline src/metrocs/redistricting/StarGeneratorTest.java:20]: Confusing variable names. The use of "...Location" suggests that the variable is holding a reference to a Location object, but that isn't the case.

What is the intent of the initializations? The values assigned don't appear to be used anywhere.

File: src/metrocs/redistricting/StarGeneratorTest.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        //check if method returns null
+        assertNotNull(StarGenerator.generatePattern(1));
+        assertNotNull(StarGenerator.generatePattern(4));
+        //assign same parameter, test if same result
+        firstLocation = StarGenerator.generatePattern(4);
+        secondLocation = StarGenerator.generatePattern(4);
+        assertEquals(firstLocation,secondLocation);
```
Comment (jody) [inline src/metrocs/redistricting/StarGeneratorTest.java:27]: Why should these return "equal" values?
Where in the documentation is it specified that elements be in the same order in the ArrayList objects?

**Review by jody — CHANGES_REQUESTED** [review]

**Comment by DanielShermanSchool** [comment]
Updated class to remove ambiguous tests, we're focusing on having a test foundation with a simple testing method to interface with JUnit.

File: src/metrocs/redistricting/StarGeneratorTest.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+/**
+ * Tests for Voter objects.
+ *
+ * @author Jada, David, Colin, John, Daniel, Ali
+ * @version
+ */
+public class StarGeneratorTest {
+    @Test
+    public void generatePatternTest() {
+        //check if method returns null
```
Comment (jody) [inline src/metrocs/redistricting/StarGeneratorTest.java:14]: This is a vacuous comment in that it doesn't add any additional explanation to the code that isn't obvious by reading the code itself. The information delivered by this comment should be **why** this is checking whether or not the method returns null.

File: src/metrocs/redistricting/StarGeneratorTest.java [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+import java.util.ArrayList;
+/**
+ * Tests for Voter objects.
+ *
+ * @author Jada, David, Colin, John, Daniel, Ali
+ * @version
+ */
+public class StarGeneratorTest {
+    @Test
+    public void generatePatternTest() {
```
Comment (jody) [inline src/metrocs/redistricting/StarGeneratorTest.java:13]: This method name indicates that this is a more complete test of the method. Perhaps it should be renamed to be more specific as to what is being tested/affirmed.

**Review by jody — CHANGES_REQUESTED** [review]
Overall looks fine, just a problem with the internal code comment which might also reflect a problem with the name of the test method.

**Comment by jody** [comment]
No update in over 30 days. Changing status to draft.

If no further updates in the next 14 days or if new PR addresses issue will delete.

**Pull request closed without merging**